api/src/controllers/users.ts

```diff
@@ -142,4 +147,92 @@ export default class UsersController {
 
         return res.status(204).json({ msg: "Logged out successfully." });
     }
+
+    public async getFriends(req: Request, res: Response) {
+        const userRepository = this.repository;
+        const friendshipRepository = this.friendshipRepository;
+
+        const userId = req.params.id;
+
+        const friendships = await friendshipRepository.find({ user_id1: userId });
+
+        if (friendships.length === 0) {
+            return res.status(200).json([]);
+        }
+
+        const friends = await userRepository.findByIds(friendships.map(friendship => friendship.user_id2));
+        return res.status(200).json(friends);
+    }
+
+    public async acceptFriend(req: Request, res: Response) {
+        const friendshipRepository = this.friendshipRepository;
+
+        const user1Id = Number(req.params.id);
+        const user2Id = Number(req.body.friendId);
+
+        const friendship = await friendshipRepository.findOne({ where: { user_id1: user1Id, user_id2: user2Id } });
+
+        if (friendship.status !== FriendshipStatus.PENDING) {
+            throw new Error('Friendship not pending');
+        }
+
+        friendship.status = FriendshipStatus.ACCEPTED;
+        await friendshipRepository.save(friendship);
+    }
+
+    public async addFriend(req: Request, res: Response) {
+        const userRepository = this.repository;
```

Comment:
The same as the other one, this probably should use the userId from the auth token.